### PR TITLE
fix: Updating shrev to 0.8

### PR DESCRIPTION
**Pull request by Rhuagh**
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/511)

### CODE REVIEW

File: src/app.rs [hunk 1]
```diff
@@ -158,13 +158,11 @@ impl<'a, 'b> Application<'a, 'b> {
...
-                .lossy_read(&mut self.events_reader_id)
-            {
-                Ok(data) => data.cloned().collect(),
-                _ => Vec::default(),
-            };
+                .read(&mut self.events_reader_id)
+                .cloned()
+                .collect::<Vec<_>>();
 
             for event in events {
```
Comment (Aceeri) [inline src/app.rs]: Maybe use the `StorageIterator` that the event channel gets in the loop below?

File: amethyst_input/src/system.rs [hunk 2]
```diff
@@ -53,11 +53,8 @@ where
     );
 
     fn run(&mut self, (input, mut handler, mut output): Self::SystemData) {
-        match input.lossy_read(&mut self.reader) {
-            Ok(data) => for d in data {
-                Self::process_event(d, &mut *handler, &mut *output);
-            },
-            _ => (),
+        for d in input.read(&mut self.reader) {
```
Comment (Aceeri) [inline amethyst_input/src/system.rs]: Rename `d` -> `data` or maybe `event` or something?

File: src/app.rs (hunk 1 above)
Comment (Rhuagh) [reply src/app.rs]: Can't because of borrow rules on world

Comment (Aceeri) [reply src/app.rs]: Ah true

**Comment by Xaeroxe** [comment]
:lgtm:

---

Review status: 0 of 12 files reviewed at latest revision, 2 unresolved discussions.

---

*[amethyst_input/src/system.rs, line 56 at r1](https://reviewable.io:443/reviews/amethyst/amethyst/511#-L0aIhTBEkmCJD9dH_cO-r1-56:-L0aw2W0cGB167d_xlRC:ba12qre) ([raw file](https://github.com/amethyst/amethyst/blob/a5dbe33bb8d61a051b03464bee6d93f6eb26bfdd/amethyst_input/src/system.rs#L56)):*
<details><summary><i>Previously, Aceeri (Aceeri) wrote…</i></summary><blockquote>

Rename `d` -> `data` or maybe `event` or something?
</blockquote></details>

My vote is for event.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/amethyst/amethyst/511#-:-L0aw6QAt0WGakOqLltt:bnfp4nl)*

**Comment by Xaeroxe** [comment]
Thanks @Rhuagh this looks great!

**Comment by Rhuagh** [comment]
Review status: 0 of 12 files reviewed at latest revision, 2 unresolved discussions.

---

*[amethyst_input/src/system.rs, line 56 at r1](https://reviewable.io:443/reviews/amethyst/amethyst/511#-L0aIhTBEkmCJD9dH_cO-r1-56:-L0desNGJF2C9vjSh7iX:b-896fix) ([raw file](https://github.com/amethyst/amethyst/blob/a5dbe33bb8d61a051b03464bee6d93f6eb26bfdd/amethyst_input/src/system.rs#L56)):*
<details><summary><i>Previously, Xaeroxe (Jacob Kiesel) wrote…</i></summary><blockquote>

My vote is for event.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/amethyst/amethyst/511)*

**Comment by torkleyy** [comment]
Reviewed 11 of 12 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/amethyst/amethyst/511)*

**Pull request merged**